### PR TITLE
Feat/adding typescript

### PR DESCRIPTION
**Pull request by emanuelbodin**
Review and release. Package can then be deprecated.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 npm install
 npm test
-rm -rf node_modules
\ No newline at end of file
+tsc
+#rm -rf node_modules
```
Comment (emanuelbodin) [inline build.sh]: Unsure if this line is needed

File: package.json [hunk 2]
```diff
@@ -48,23 +48,16 @@
     "node-notifier": "^10.0.1",
     "prettier": "^2.8.3",
     "pretty-quick": "^3.1.3",
-    "set-value": "^4.1.0"
+    "set-value": "^4.1.0",
+    "ts-jest": "^29.0.5",
+    "typescript": "^4.9.5",
+    "mongoose": "^7.0.0"
```
Comment (emanuelbodin) [inline package.json]: Need to have mongoose as devDependency to make typescript happy

File: package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "@kth/log": "^4.0.5"
   },
   "peerdependencies": {
-    "mongoose": "^6.6.0"
+    "mongoose": "^7.0.0"
```
Comment (emanuelbodin) [inline package.json:33]: Updated mongoose to v7 since it provides types, then we don't need to install @types/mongoose

File: src/index.ts [hunk 4]
```diff
@@ -69,14 +74,10 @@ async function connect(options) {
       log.fatal('DATABASE: Connection error', { error })
       Global.isConnected = false
     })
-
-    return mongoose.connect(dbUri, dbOptions).then(data => {
```
Comment (emanuelbodin) [inline src/index.ts]: I do not think need to return `mongoose.connect` here

File: src/index.ts [hunk 5]
```diff
@@ -13,31 +9,40 @@ const standardOptions = {
...
+  isConnected: undefined,
   checkConnectionOnceTimerID: null,
 }
 
-function _getMongoOptionsWithoutDbUri(options) {
-  const dbOptions = { ...standardOptions, ...options }
-  delete dbOptions.dbUri
-  delete dbOptions.logger
-  return dbOptions
+type ConnectOptions = {
```
Comment (emanuelbodin) [inline src/index.ts:22]: I am unsure of what `ConnectOptions ` should look like, maybe add properties here when we try this package out in a project?

File: src/index.ts [hunk 6]
```diff
@@ -13,31 +9,37 @@ const standardOptions = {
...
```
Comment (nsandstrom) [inline src/index.ts]: Det finns en "type ConnectOptions" även i mongoose-paketet. Men dom har olika innehåll och används för olika syften. Blir lätt förvirrande.

File: src/index.ts [hunk 7]
```diff
@@ -13,31 +9,37 @@ const standardOptions = {
   // heartbeatFrequencyMS: 5000,
 }
 
-const Global = {
-  isConnected: null,
+type TGlobal = {
```
Comment (nsandstrom) [inline src/index.ts]: Global (and TGlobal) are unclear names. It is used as a collection of status flags?

Also it looks like `checkConnectionOnceTimerID` is never used?
Can this just be replaced by `const isConnected` instead?

File: src/index.ts [hunk 8]
```diff
@@ -13,31 +9,37 @@ const standardOptions = {
...
-function _getMongoOptionsWithoutDbUri(options) {
-  const dbOptions = { ...standardOptions, ...options }
-  delete dbOptions.dbUri
-  delete dbOptions.logger
-  return dbOptions
+type ConnectOptions = {
+  logger?: null
+  dbUri: string
+  mongooseDebug?: boolean
 }
```
Comment (nsandstrom) [inline src/index.ts]: user, pass and ssl are now missing

File: src/index.ts [hunk 9]
```diff
@@ -13,31 +9,37 @@ const standardOptions = {
...
 
-/**
- * @returns {boolean}
- *      True iff default connection to MongoDB is currently established
- */
-function isOk() {
+function _getMongoOptionsWithoutDbUri(options: ConnectOptions) {
+  const { mongooseDebug } = options
```
Comment (nsandstrom) [inline src/index.ts]: user, pass and ssl are now missing

File: src/index.ts (hunk 7 above)
Comment (emanuelbodin) [reply src/index.ts]: Seems like you are right

File: src/index.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
+import kthLog from '@kth/log'
+import mongoose, { type ConnectOptions } from 'mongoose'
+
+let isConnected = false
+
+// True if default connection to MongoDB is currently established
+export function isOk() {
+  return isConnected === true
```
Comment (nsandstrom) [inline src/index.ts:8]: https://masteringjs.io/tutorials/mongoose/connection-status

Not sure if the link above is still relevant, but we might be able to use `mongoose.connection.readyState == 1` instead and drop `isConnected`

**Pull request closed without merging**